### PR TITLE
LPS-204337  Enable Navigation to the Options of the Panel Link and Account Restricted Field Using the Arrow Keys

### PR DESCRIPTION
**Pull request by igor-franca**
Related Issue: [LPS-204337](https://liferay.atlassian.net/browse/LPS-204337)

[previous review](https://github.com/liferay-objects/liferay-portal/pull/2978)

### CODE REVIEW

**Comment by rebsilva** [comment]
this will be reviewed after the model builder prioritization bugs

**Comment by marcelabc** [comment]
Hey @danvb112, 
can you help and review this one?
I believe this would help a lot @rebsilva because she had some PRs to review :heart_decoration:

**Comment by rebsilva** [comment]
i think maybe i'll have some time at the end of the day to review this, i'll let you know about it @danvb112; also thanks a lot @marcelabc for the help and concern! :purple_heart:

File: modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ObjectDetails/ScopeContainer.tsx [hunk 1]
```diff
@@ -104,52 +104,53 @@ export function ScopeContainer({
 						});
 					}
 
-					setSelectedPanelCategoryValue('');
+					setSelectedPanelCategoryValue(undefined);
 				}}
 				selectedKey={values.scope}
 			/>
-
```
Comment (danvb112) [inline modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ObjectDetails/ScopeContainer.tsx:111]: Hey @igor-franca, I encountered a bug when trying to change the scope and link category of the panel multiple times using the keyboard, unfortunately this is a common problem with the Picker component, the best way to resolve this is to quickly remove the component from the screen and place it again, forcing the component to rerender. 

So I suggest you create a new state (`loadingPanelLinkSelect` for example) at the top of the component, and in the existing useEffet you can manage the state:


```tsx
	useEffect(() => {
		setPanelCategoryKey(
			values.scope === 'company' ? companies : sites,
			values.panelCategoryKey as string
		);

		setLoadingPanelLinkSelect(true);

		setTimeout(() => setLoadingPanelLinkSelect(false), 100);
		// eslint-disable-next-line react-hooks/exhaustive-deps
	}, [values.scope, companies, sites]);
```

then add the state to make the condition here and place the loading indicator.

**Review by danvb112 — CHANGES_REQUESTED** [review]
Hey @igor-franca, I have one suggestion, fell free to ask me anything :smile:

**Comment by igor-franca** [comment]
ci:test:sf

**Comment by igor-franca** [comment]
ci:stop

**Comment by igor-franca** [comment]
Hey, @danvb112. After clarifying some [doubts](https://liferay.slack.com/archives/C049XG98GQL/p1707334836231499) with the platform experience team, we discovered that it was a bug in Clay. And a fix has just been submitted. Therefore, this will be available in the next release. Does it make sense to add this workaround to the master? Since it will be available in the next release along with the Clay fix?

**Comment by igor-franca** [comment]
As discussed [here](https://liferay.slack.com/archives/C049XG98GQL/p1707334836231499), an issue was identified in Clay, which has now been resolved. After the release of the next release of clay, this behavior should be fixed.

**Pull request closed without merging**